applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -148,8 +148,8 @@ static void terminate_dfu(void)
 	__ASSERT_NO_MSG(flash_area != NULL);
 
 	flash_area_close(flash_area);
-	k_delayed_work_cancel(&dfu_timeout);
```

Comment:
Ok. Cancels will not race with any resubmit - all operations are in sync with system workqueue.